src/radical/entk/execman/rp/task_processor.py

```diff
@@ -177,32 +177,19 @@ def resolve_arguments(args, placeholders, logger):
 #
 def resolve_tags(task, parent_pipeline_name, placeholders):
 
-    # entk only handles co_location tags.  If tags are given as strings, they
-    # get translated into `{'colocate': '<tag>'}`.  Tags passed as dictionaries
-    # are checked to conform with above form.
-    #
-    # In both cases, the tag string is expanded with the given placeholders.
+    tags = dict(task.tags or {'colocate': task.uid})
```

Comment:
Am I reading this right that the `colocate` tag is _*always*_ specified by default?  That's not part of the PR, but surprising nevertheless.  From the comment, I would have expected the code to look more like
```py
if isinstance(task.tags, str):
    tags = {'colocate': task.tags}
elif isinstance(task.tags, dict):
    tags = task.tags
else:
    tags = dict()
```
or something like that.  Since the `colocate` list's the task's _*own*_ uid, what is the expected scheduler semantics for that tag?

Reply:
> I would have expected the code to look more like

yeah, that comments are confusing, that's why I removed them :) as of Task's description the type of `tags` is always `dict`, thus it is either provided as a dict or created here with task's `uid`

> what is the expected scheduler semantics for that tag?

since every task will have its own tag, then it is the same as no tasks have any tag. Honestly I didn't try to understand why tags were enforced..